rr_computer_vision/include/lane_detection.hpp

```diff
@@ -6,22 +6,71 @@
 #ifndef __LANE_DETECTION_HPP
 #define __LANE_DETECTION_HPP
 
-
 //ROS
 #include <ros/ros.h>
+#include <sensor_msgs/Image.h>
 
 //OPENCV includes
 #include <opencv2/opencv.hpp>
 
+#include <cv_bridge/cv_bridge.h>
+#include <nav_msgs/OccupancyGrid.h>
+
 //LOCAL
 
 class LaneDetection
 {
-  private:
-
   public:
-    LaneDetection();
+    LaneDetection(ros::NodeHandle nh);
     ~LaneDetection();
+    void Multithreshold(const cv::Mat &input_image, const cv::Mat &bounds, cv::Mat &output_image);
+    void FindWhite(const cv::Mat &input_image, const cv::Scalar bounds, int patch_size, cv::Mat &output_image);
+    cv::Mat GetContours(const cv::Mat &image, int min_size);
+    
+  private:
+    void InitializeSubscribers();
+    void InitializePublishers();
+    void RGBCameraCallback(const sensor_msgs::Image& msg);
+
+    int adapt_hsv_patch_size_;
+    int blob_size_;
+
+    ros::Subscriber zed_subscriber;
+    ros::Publisher test_publisher;
+    ros::Publisher pointList_pub_;
+    ros::NodeHandle nh_;
+
+    cv::Mat im_input_;
```

Comment:
Are you sure you need all of these variables to be global within the class?
Is there no nice way of making some of these local variables instead?
As long as you understand what comes in and out of the function this should be a relatively easy modification to do.